Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java

```diff
@@ -400,7 +400,19 @@ private Mets toMets() {
         LinkedList<Pair<String, String>> smLinkData = new LinkedList<>();
         StructMapType logical = new StructMapType();
         logical.setTYPE("LOGICAL");
-        logical.setDiv(new DivXmlElementAccess(workpiece.getStructure()).toDiv(mediaUnitIDs, smLinkData, mets));
+        DivType structureRoot = new DivXmlElementAccess(workpiece.getStructure()).toDiv(mediaUnitIDs, smLinkData, mets);
+        List<LinkedStructure> uplinks = workpiece.getUplinks();
+        if (uplinks.isEmpty()) {
+            logical.setDiv(structureRoot);
+        } else {
+            DivType uplinkHolder = new DivType();
+            Mptr uplink = new Mptr();
+            uplink.setHref(uplinks.get(uplinks.size() - 1).getUri().getPath());
+            uplinkHolder.getMptr().add(uplink);
+            uplinkHolder.getDiv().add(structureRoot);
+            logical.setDiv(uplinkHolder);
```

Comment:
setDiv does not sound like  method to put an uplinkHolder...
logical is from type "structMapType" why can I set a div on a type?

Reply:
I find that confusing, too. `logical` is the logical `<structMap>`. It is of the “type” structMap (for JAXB, _type_ seems to mean _the name of the XML element_), so JAXB calls the corresponding class `StructMapType`. This does **not refer to** the attribute `TYPE` (=`LOGICAL`) on the XML element.  So I am setting the `<div>` into the `<structMap>` here, as common.

Now for the link: Without link, in `<structMap>` is the `<div>`, which represents the root of the structure node (`structureRoot`). If there is a link up, a meaningless `<div>` is first inserted in `<structMap>`, which only serves as a holder for the `<mptr>` with the link and `structureRoot`. So here `uplinkHolder` is also a `<div>`.